### PR TITLE
sdk update to v0.11.0

### PR DESCRIPTION
**Pull request by grvsahil**
### Description

This PR includes the following changes:

1. conduit-connector-sdk updated (v0.9.1 => v0.11.0).
2. Used "paramgen" to generate config parameters.
3. Renamed config package to "common".

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-s3/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,3 +89,7 @@ the differentiation. So we consider all the objects in the bucket irrespective o
 
 If a pipeline restarts during the snapshot, then the connector will start scanning the objects from the beginning of
 the bucket, which could result in duplications.
+
+### Useful resources
```
Comment (maha-mstfa) [inline README.md]: do we have any resources? it's an empty header

File: source/source.go [hunk 2]
```diff
@@ -20,71 +20,53 @@ import (
...
+func (s *Source) Configure(ctx context.Context, cfgRaw commonsConfig.Config) error {
 	logger := sdk.Logger(ctx).With().Str("Class", "Source").Str("Method", "Configure").Logger()
 	logger.Trace().Msg("Starting Configuring the Source Connector...")
 
-	sourceConfig, err := srcConfig.ParseSourceConfig(ctx, cfg)
+	err := sdk.Util.ParseConfig(ctx, cfgRaw, &s.config, NewSource().Parameters())
 	if err != nil {
-		logger.Error().Stack().Err(err).Msg("Error While parsing the Source Config")
-		return err
+		return err //nolint: wrapcheck // not needed here
```
Comment (maha-mstfa) [inline source/source.go]: no harm in wrapping it here, imo it's better to, just a simple `fmt.Errorf("invalid config: %w", err)`

File: README.md (hunk 1 above)
Comment (grvsahil) [reply README.md]: yes right, removed it

**Review by maha-mstfa — APPROVED** [review]
nice work!

**Pull request merged**